### ISSUE TITLE
Xcake creates non-functional unit test target

### ISSUE THREAD

**Issue by maximkhatskevich** [jcampbell05/xcake#120]
This particular issue happens when host target is a framework. As it has been reported a long time ago, Xcake does not connect properly host targets with their unit tests target(s), so unit tests are not working out of the box after generating project file with Xcake.

Checkout code from this branch https://github.com/maximkhatskevich/MKHOperationFlow/tree/xcake-broken-tests then run "xcake make" in root folder and then open "MKHOperationFlow.xcworkspace" file and hit "Cmd+U" hotkey.

**Expected behavior:** hit "Cmd+U" hotkey in Xcode and unit tests will be executed.
**Actual behavior:** nothing happens at all.

P.S. This issue might be related to issues https://github.com/jcampbell05/xcake/issues/114 and https://github.com/jcampbell05/xcake/issues/115 , but not sure.

**Comment by maximkhatskevich** [comment]
The only workaround for now is to remove created by Xcake unit tests target and create one manually using standard Xcode functionality, then include unit tests source files into the newly created target. After this, everything works properly.

**Comment by jcampbell05** [comment]
So if host target isn't an app it breaks ?

**Comment by maximkhatskevich** [comment]
It doesn't work anyway out of the box, Xcake does not connect unit test target to the host target, so when you hit "Cmd+U" nothing happens. As a workaround - I have to go manually into the host target scheme and in the "Test" section add unit tests target to the list. After this, if host target is an app - it starts working. If host target is a framework - it still does not work, giving me a strange error message when try to run tests. So the only workaround for a framework host target - remove test target at all and create new one manually, so Xcode can configure everything correctly for me. See the sample code to reproduce the issue. Hope that makes sense. Lmk if you have questions.

**Comment by chiahsien** [comment]
I have same issue, here is my Cakefile.

```ruby
project.name = "iCHEF"
project.class_prefix = "iCHEF"
project.organization = "iCHEF"

application_for :ios, 8.1 do |target|
    target.name = "iCHEF"
    target.language = :objc
    target.include_files = ["iCHEF/Src/**/*.*"]

    unit_tests_for target do |test_target|
      test_target.name = "iCHEFKiwiTests"
      test_target.include_files = ["iCHEFKiwiTests/Unit Test/**/*.*"]
    end
end
```

As you can see, it's very simple, I just create an app target and an unit test target. But the unit test target fails to connect its host to the app target.

<img width="730" alt="2016-11-18 12 09 46" src="https://cloud.githubusercontent.com/assets/1225437/20418538/61539c1c-ad89-11e6-8bb5-28a306700214.png">

I've tried to set unit test's `TEST_HOST` configuration to `$(BUILT_PRODUCTS_DIR)/iCHEF.app/iCHEF`, still not working.

Any idea?

Xcake: 0.7.1
Xcodeproj: 1.3.3
Xcode: 8.0

**Comment by jcampbell05** [comment]
@maximkhatskevich I've fixed this in 0.8.0 the issue was xcodeproj wasn't setting the search path so when you were unit testing a framework the framework wasn't loaded. 

It should be fixed now :)
